### PR TITLE
style: Sort package lists, configure options, and other various sortable files

### PR DESCRIPTION
**Pull request by echoix**
Follow up from last month's #4373.

This PR sorts sortable parts of many remaining files. Build scripts, package lists, file lists, contributors lists, etc.
I made some searches on what looked like configure flags, and packages too.

About the only thing that wasn't sorted was everything in Makefiles, like directory lists or other lists in Makefiles, and the mingw64 packages in the osgeo4w.yml workflow, as the packages weren't placed in a line-delimited list yet

### CODE REVIEW

File: rpm/grass.spec [hunk 1]
```diff
@@ -37,64 +37,66 @@ BuildRequires:	flexiblas-devel
...
-BuildRequires:	python%{python3_version_nodots}-numpy
-%else
-BuildRequires:	python3-numpy
-%endif
-%if 0%{?rhel} && 0%{?rhel} == 7
-BuildRequires:	postgresql-devel
+BuildRequires:	python%{python3_version_nodots}-dateutil
 %else
-BuildRequires:	libpq-devel
+BuildRequires:	python3-dateutil
```
Comment (neteler) [inline rpm/grass.spec:79]: I am not sure about the changes in this SPEC file, also in the lines above and below.

Note that the upstream file is here (once settled here I can fwd-port desired changes):
https://src.fedoraproject.org/rpms/grass/blob/rawhide/f/grass.spec

**Comment by neteler** [comment]
Thanks for your efforts.

We have a number of CSV files, perhaps some could be merged? Their origin is from CVS/SVN times with a different than GitHub write access model which is no longer in place:

- `contributors.csv`
- `contributors_extra.csv`
- `translators.csv`
- `utils/git_author_github_name.csv`
- `utils/svn_name_git_author.csv`
- `utils/svn_name_github_name.csv`

In case of structural changes (e.g., merging selectively) also a GUI change would be needed:

`wxpython/gui_core/ghelp.py`

```python
            contribfile = os.path.join(os.getenv("GISBASE"), "contributors_extra.csv")
            contribfile = os.path.join(os.getenv("GISBASE"), "contributors.csv")
        translatorsfile = os.path.join(os.getenv("GISBASE"), "translators.csv")
                label=_("%s file missing") % "translators.csv",
```

However, maybe better done in a separate PR.

File: rpm/grass.spec (hunk 1 above)
Comment (echoix) [reply rpm/grass.spec:79]: If the order of the dependencies (for a given section) doesn't matter, it is equivalent and no lines are lost, and the same is in the if else blocks. If the order matters for example on their side knowing what package has to be built before (at an ecosystem level), then we might need to revert the changes here.

**Comment by echoix** [comment]
> Thanks for your efforts.
> 
> We have a number of CSV files, perhaps some could be merged? Their origin is from CVS/SVN times with a different than GitHub write access model which is no longer in place:
> 
> * `contributors.csv`
> * `contributors_extra.csv`
> * `translators.csv`
> * `utils/git_author_github_name.csv`
> * `utils/svn_name_git_author.csv`
> * `utils/svn_name_github_name.csv`
> 
> In case of structural changes (e.g., merging selectively) also a GUI change would be needed:
> 
> `wxpython/gui_core/ghelp.py`
> 
> ```python
>             contribfile = os.path.join(os.getenv("GISBASE"), "contributors_extra.csv")
>             contribfile = os.path.join(os.getenv("GISBASE"), "contributors.csv")
>         translatorsfile = os.path.join(os.getenv("GISBASE"), "translators.csv")
>                 label=_("%s file missing") % "translators.csv",
> ```
> 
> However, maybe better done in a separate PR.

I made a separate issue for it https://github.com/OSGeo/grass/issues/4599, as indeed, it is kind of a different topic than this PR.

File: rpm/grass.spec (hunk 1 above)
Comment (neteler) [reply rpm/grass.spec:79]: Right. I have extracted `grass.spec` from this PR and built it locally with Fedora tools, all fine.
Hence updated accordingly https://src.fedoraproject.org/rpms/grass/blob/rawhide/f/grass.spec

My suggestion would be to merge that SPEC file back into this PR.

Comment (echoix) [reply rpm/grass.spec:79]: I merged back the spec file, keeping the 8.4.0 release 3 version, as updating it to 8.5 would probably need a changelog entry, and isn't synced back to rawhide.

Comment (neteler) [reply rpm/grass.spec:79]: Great, thanks a lot.

**Review by neteler — APPROVED** [review]
The changes look good to me.

**Pull request merged**